libgnucash/engine/test/gtest-gnc-datetime.cpp

```diff
@@ -182,7 +221,7 @@ TEST(gnc_date_constructors, test_str_format_constructor)
             got_month = test_ymd.month;
             got_day = test_ymd.day;
         }
-        catch (const std::invalid_argument& e)
+        catch (...)
```

Comment:
I'd rather have our tests verify the exact exceptions to be thrown so they would fail if new exceptions crop up. I presume boost throws a different exception ? The examples failing boost tests could set the result values to -2, so we can test both exceptions.

Even better though would be to fix the source code such that only one exception type is thrown. You can do that by catching the boost exceptions and rethrowing them as std::invalid_argument.